dsgrid/project.py

```diff
@@ -230,18 +237,42 @@ def _build_filtered_record_ids_by_dimension_type(self, context: QueryContext):
                     "id"
                 )
             else:
+                query_name = dim_filter.dimension_query_name
+                records = self._config.get_dimension_records(query_name)
+                df = dim_filter.apply_filter(records).select("id")
                 supp_query_names = set(
                     self._config.list_dimension_query_names(
                         category=DimensionCategory.SUPPLEMENTAL
                     )
                 )
-                records = self._config.get_dimension_records(dim_filter.dimension_query_name)
-                df = dim_filter.apply_filter(records).select("id")
-                query_name = dim_filter.dimension_query_name
                 if query_name in supp_query_names:
-                    mapping_records = self._config.get_base_to_supplemental_mapping_records(
-                        query_name
+                    assert isinstance(dim_filter, DimensionFilterSingleQueryNameBaseModel)
+                    supp_dim = self._config.get_dimension(query_name)
+                    mapping_configs = self._config.list_base_to_supplemental_mapping_configs(
+                        supplemental_dimension_id=supp_dim.model.dimension_id,
                     )
+                    mapping_config: MappingTableConfig
+                    if len(mapping_configs) > 1:
+                        if dim_filter.base_dimension_query_name is None:
+                            msg = (
+                                "There is more than one base-to-supplemental mapping config for "
+                                f"{supp_dim.model.label}. Please specify base_dimension_query_name "
```

Comment:
Is it possible to know the base_dimension_query_name based on what datasets are being combined, rather than requiring it as user input?

Reply:
I added a new rule that says all datasets in a query have to be tied to the same project base dimensions. I record the base dimension names in the QueryContext, so we have access to it here and everywhere else in the query process. Please raise a concern if you think a query needs to support datasets mapped to different project base dimensions. That should be handled by our to-be-written code that combines datasets.

Reply:
I like this as a way to reduce complexity and allow us to move forward. I also agree that perhaps all such combinations that violate this rule will be handled by the to-be-written combine-datasets code (i.e., not in "query").